src/v/security/gssapi_principal_mapper.cc

```diff
@@ -377,21 +371,17 @@ ss::sstring gssapi_rule::replace_substitution(
 }
 
 gssapi_principal_mapper::gssapi_principal_mapper(
-  std::string_view default_realm,
   config::binding<std::optional<std::vector<ss::sstring>>>
     principal_to_local_rules_cb)
-  : _default_realm(default_realm)
-  , _principal_to_local_rules_binding(std::move(principal_to_local_rules_cb))
-  , _rules{detail::parse_rules(
-      _default_realm, _principal_to_local_rules_binding())} {
+  : _principal_to_local_rules_binding(std::move(principal_to_local_rules_cb))
+  , _rules{detail::parse_rules(_principal_to_local_rules_binding())} {
     _principal_to_local_rules_binding.watch([this]() {
-        _rules = detail::parse_rules(
-          _default_realm, _principal_to_local_rules_binding());
+        _rules = detail::parse_rules(_principal_to_local_rules_binding());
```

Comment:
Oh that's a good point.  So should the mapper live within `server.cc` then and be handed off to the authenticator when necessary?

Reply:
Yeah, actually it's worth dropping the watch in this PR. Otherwise every config change requires every connection to build equivalent rules, but none of them will ever look at the rules again.

Yeah, `kafka::server` is where the `tls::principal_mapper` lives. It might be an easy change. Up to you.

Reply:
Per slack conversation I'll file a ticket to address refactoring who holds the mapper object

Reply:
https://github.com/redpanda-data/redpanda/issues/8341